scripts/are_queued_jobs_finished.py

```diff
@@ -45,21 +46,34 @@ def getQueuedJobIDs(jobs):
 
 def getJobStatus(job_ids):
     """ run qstat and return tuple of (job_id, status) """
-    qstat_cmd = ['qstat', '-x']
+    results = ''
+    qstat_cmd = ['qstat', '-xf']
     qstat_cmd.extend(job_ids)
-    process = subprocess.Popen(qstat_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    results = process.communicate()
+    t = TemporaryFile()
+    process = subprocess.Popen(qstat_cmd, stdout=t)
+    process.wait()
     if process.returncode != 0:
         sys.exit(1)
     else:
-        parsed_statuses = re.findall(r'(\d+).*\d+ (\w)', results[0])
-        return parsed_statuses
+        t.seek(0)
+        results = t.read()
+
+    t.close()
+    return results
 
 def updateStatus(jobs, finished_jobs):
     """ loop through jobs and adjust the status for those supplied in finished_jobs """
     for job, meta in jobs.getJobs().iteritems():
-        if 'job_id' in meta and meta['job_id'] in finished_jobs and meta['status_bucket']['status'] == 'QUEUED':
-            meta['status_bucket']['status'] = 'WAITING'
+        # Use list splicing to simplify the search process
+        if 'job_id' in meta \
+           and meta['job_id'] in finished_jobs[0::2] \
```

Comment:
Why not make `finished_jobs` a dict with `job_id` as the key and the status as the value? Then you wouldn't have to do the stride splicing.

Reply:
I thought about doing it that way in the beginning... I'll take another stab at it. But I hit a road block somewheres...